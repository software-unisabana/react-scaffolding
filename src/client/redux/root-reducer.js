import { combineReducers } from 'redux';

import itemReducer from './item/item.reducer';

const rootReducer = combineReducers({
	products: itemReducer,
});

export default rootReducer;
