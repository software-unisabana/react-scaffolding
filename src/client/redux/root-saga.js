import { all, call } from 'redux-saga/effects';

import { itemSagas } from './item/item.sagas';

export default function* rootSaga() {
	yield all([call(itemSagas)]);
}
