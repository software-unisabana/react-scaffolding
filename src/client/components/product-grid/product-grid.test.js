import React from 'react';
import { mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ProductGrid from './product-grid.component';
import ProductPreview from '../product-preview/product-preview.component';

export const createMockStore = ({ state, reducers }) => {
	const store = createStore(combineReducers(reducers), state);
	return {
		...store,
	};
};

describe('ProductGrid Component', () => {
	let wrapper;
	let store;
	beforeEach(() => {
		const mockReducer = (
			state = {
				isFetching: true,
				items: [],
			},
			action
		) => state;
		const mockState = {
			products: {
				items: [
					{
						id: 'MLA689325634',
						title: 'Ps5',
						price: {
							currency: 'ARS',
							amount: 38500,
							decimals: 2,
						},
						picture: '',
						condition: 'new',
						free_shipping: false,
						sold_quantity: 5,
						description: 'ps%',
					},
				],
				isFetching: false,
			},
		};
		store = createMockStore({
			state: mockState,
			reducers: { products: mockReducer },
		});
		wrapper = mount(
			<BrowserRouter>
				<Provider store={store}>
					<ProductGrid />
				</Provider>
			</BrowserRouter>
		);
	});
	it('should render ProductPreview component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should render an equal number of ProductPreview components as the Items prop', () => {
		expect(wrapper.find(ProductPreview).length).toEqual(1);
	});
});
