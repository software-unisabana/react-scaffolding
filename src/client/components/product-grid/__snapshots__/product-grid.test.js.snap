// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid Component should render ProductPreview component 1`] = `ReactWrapper {}`;
